src/mlpack/methods/kde/kde_rules_impl.hpp

```diff
@@ -200,10 +322,111 @@ Score(TreeType& queryNode, TreeType& referenceNode)
           referenceNode.NumDescendants() * kernelValue;
```

Comment:
You're right. It was probably more difficult to think about it in #1301 because in that PR the rules didn't try to approximate self-child, so this wasn't necessary. Thanks for pointing it out :)